### ISSUE TITLE
Fatal DirectX View instancing assertion failure

### ISSUE THREAD

**Issue by moleium** [GaijinEntertainment/DagorEngine#104]
East district example is completely broken on relatively old Intel integrated graphics.
Assertion in [render_state.h](https://github.com/GaijinEntertainment/DagorEngine/blob/main/prog/engine/drv/drv3d_DX12/render_state.h#L403) assumes every machine has a GPU with view instancing support:
```
assert failed: "result.viewInstanceCount == 0 || d3d::get_driver_desc().caps.hasBasicViewInstancing"
```

Fails on Intel HD Graphics 520 which **DOESN'T SUPPORT** view instancing, despite having DX12 FL 12.1. Either.

I confirm that it does work after switching to DX11 in the settings datablock of the scene. although,
There should be a check for hardware support and gracefully fallback to older DirectX API.

Environment:
- GPU: Intel HD Graphics 520
- Driver: 30.0.100.9865
- DX12 Feature Level: 12.1

**Comment by NicSavichev** [comment]
Thanks for findings, forwared issue to our graphics programmers and will come back with an answer later.

**Comment by NicSavichev** [comment]
Can you build dngSceneViewer from source code (to have PDB locally) and post callstack of that assert?
We have suspection that some corruption occurs, code should not read this assert at all

**Comment by NicSavichev** [comment]
> Can you build dngSceneViewer from source code (to have PDB locally) and post callstack of that assert?

Or I can send you PDB for EXE distributed in prebuilt version (only 2024/12/17 one)

**Comment by NicSavichev** [comment]
Should be fixed in https://github.com/GaijinEntertainment/DagorEngine/commit/0f593367905adc2332f771e6b94c403600f787e3
But one needs to rebuild dngSceneViewers from source code to test on older Intel HD (to confirm assert has gone)

**Comment by moleium** [comment]
> Should be fixed in [0f59336](https://github.com/GaijinEntertainment/DagorEngine/commit/0f593367905adc2332f771e6b94c403600f787e3) But one needs to rebuild dngSceneViewers from source code to test on older Intel HD (to confirm assert has gone)

I'm on a limited bandwidth connection and storage space. Building from source isn't feasible for me right now. Could you either:
- Share the PDB for the prebuilt version
- Or provide a prebuilt binary with the fix from the latest commit?

I can help verify the fix once I have either of those. Thanks for the quick responses by the way.

**Comment by NicSavichev** [comment]
> > Should be fixed in [0f59336](https://github.com/GaijinEntertainment/DagorEngine/commit/0f593367905adc2332f771e6b94c403600f787e3) But one needs to rebuild dngSceneViewers from source code to test on older Intel HD (to confirm assert has gone)
> 
> I'm on a limited bandwidth connection and storage space. Building from source isn't feasible for me right now. Could you either:
> 
> * Share the PDB for the prebuilt version
> * Or provide a prebuilt binary with the fix from the latest commit?
> 
> I can help verify the fix once I have either of those. Thanks for the quick responses by the way.

[Uploading dng_scene_viewer-dev.zip…]()
Here it is - windows-x86_64 EXE built from repo master
